addons/web/static/src/js/views/basic/basic_model.js

```diff
@@ -1284,14 +1289,27 @@ var BasicModel = AbstractModel.extend({
                 // optionally clear the DataManager's cache
                 self._invalidateCache(parent);
                 if (!_.isEmpty(action)) {
-                    return self.do_action(action, {
-                        on_close: function () {
-                            return self.trigger_up('reload');
-                        }
+                    return new Promise(function (resolve, reject) {
+                        self.do_action(action, {
+                            on_close: function (result) {
+                                return self.trigger_up('reload', {
+                                    onSuccess: resolve,
+                                });
+                            }
+                        });
```

Comment:
@msh-odoo Can you explain to me what motivates this change ? I don't really know the changes that happened in the past when I was not at Odoo, so this feels weird. What's happening ?

Reply:
@SimonGenin Previously when we Archive/Unarchive record it just calls archive/unarchive method of model but after commit: 71d4116ea379f204d5773d829dce624fbcf67ec5 ```action_archive``` method now return action as well, say for example ```hr.employee``` when you archive 1 record from listview then it will open another wizard, now on close of that wizard we want to refresh listview to go back to previous page, so to handle that case, I did above code, hope this will help you to understand.